src/components/playbackManager.ts

```diff
@@ -204,46 +223,92 @@ export class playbackManager {
+        // to seconds and give it to loadRequestData :)
+        if (mediaInfo.customData.startPositionTicks > 0)
+            loadRequestData.currentTime =
+                mediaInfo.customData.startPositionTicks / 10000000;
+
         load($scope, mediaInfo.customData, item);
         this.playerManager.load(loadRequestData);
 
+        console.log('setting src to ' + streamInfo.url);
         $scope.PlaybackMediaSource = mediaSource;
-
-        console.log('setting src to ' + url);
         $scope.mediaSource = mediaSource;
+        $scope.audioStreamIndex = streamInfo.audioStreamIndex;
+        $scope.subtitleStreamIndex = streamInfo.subtitleStreamIndex;
 
         DocumentManager.setPlayerBackdrop(item);
 
-        reportPlaybackStart($scope, getReportingParams($scope));
-
         // We use false as we do not want to broadcast the new status yet
         // we will broadcast manually when the media has been loaded, this
         // is to be sure the duration has been updated in the media element
         this.playerManager.setMediaInformation(mediaInfo, false);
     }
 
-    stop(continuing = false): Promise<any> {
-        $scope.playNextItem = continuing;
-        stop();
+    /**
+     * stop playback, as requested by the client
+     */
+    static stop(): void {
+        this.playerManager.stop();
+        // onStopped will be called when playback comes to a halt.
+    }
 
-        const reportingParams = getReportingParams($scope);
+    /**
+     * Called when media stops playing.
+     * TODO avoid doing this between tracks in a playlist
+     */
+    static onStopped(): void {
+        if (this.getNextPlaybackItemInfo()) {
+            $scope.playNextItem = true;
+        } else {
+            $scope.playNextItem = false;
 
-        let promise;
+            DocumentManager.setAppStatus('waiting');
 
-        stopPingInterval();
+            stopPingInterval();
 
-        if (reportingParams.ItemId) {
-            promise = reportPlaybackStopped($scope, reportingParams);
+            DocumentManager.startBackdropInterval();
         }
+    }
 
-        this.playerManager.stop();
-
-        this.activePlaylist = [];
-        this.activePlaylistIndex = -1;
-        DocumentManager.startBackdropInterval();
+    /**
+     * Get information about the next item to play from window.playlist
+     *
+     * @returns {ItemIndex | null} item and index, or null to end playback
+     */
+    static getNextPlaybackItemInfo(): ItemIndex | null {
+        if (this.activePlaylist.length < 1) {
+            return null;
+        }
 
-        promise = promise || Promise.resolve();
+        let newIndex: number;
+
+        if (this.activePlaylistIndex < 0) {
+            // negative = play the first item
+            newIndex = 0;
+        } else
+            switch (window.repeatMode) {
+                case 'RepeatOne':
+                    newIndex = this.activePlaylistIndex;
+                    break;
+                case 'RepeatAll':
```

Comment:
IIRC, there's an enum for repeat types included in the CAF typings, at least in the sender's ones.

Reply:
We should go over both sender and receiver then..

So the sender needs to pick which version to send.

Over at this end, we can support both for a while.

However this will break the stable client, but we can luckily update that now.

Reply:
Is [this](https://github.com/hawken93/jellyfin-chromecast/pull/228/commits/8732560cb10621722eb50c013d80918c1675a614) good or does more work need to be done?